packages/integrations/useCookies/index.ts

```diff
@@ -0,0 +1,103 @@
+import { ref, getCurrentInstance, watchEffect } from 'vue-demi'
+import Cookie from 'universal-cookie'
+import type { IncomingMessage } from 'http'
+
+type RawCookies = {
+  [name: string]: string
+}
+
+let universalCookie: Cookie | undefined
+
+/**
+ * Initializing universal-cookie using request (default is window.document.cookie)
+ * @param {Object} req - incoming http request (for SSR)
+ * @see {@link https://github.com/reactivestack/cookies/tree/master/packages/universal-cookie|universal-cookie}
+ */
+export const createCookies = (req?: IncomingMessage) => {
+  universalCookie = new Cookie(req ? req.headers.cookie : null)
+
+  return universalCookie
+}
+
+/**
+ * Reactive methods to work with cookies (firstly call {@link createCookies} method if using SSR)
+ * @param {string[]|null|undefined} dependencies - array of watching cookie's names. Pass empty array if don't want to watch cookies changes.
+ * @param {boolean} [doNotParse=false] - don't try parse value as JSON
+ */
+export const useCookies = (dependencies?: string[] | null, doNotParse = false) => {
+  const cookies = universalCookie || createCookies()
+
+  let previousCookies = cookies.getAll<RawCookies>({ doNotParse: true })
+
+  /**
+   * Adds reactivity to returning methods
+   */
+  const touches = ref(0)
+
+  /**
+   * Calls inside setup() function only
+   */
+  if (getCurrentInstance()) {
```

Comment:
I think it's ok to also be called outside of `setup()`

Reply:
You can use `tryOnMounted` and `tryOnUnmounted` in utils instead. It should be fine to initialize the hook outside setup() and be reused across multiple components.

Reply:
Could you add a use case for this? `onMounted` and `onUnmounted` will work only in `setup()`, so why do I need call these methods in other places? I don't understand

Reply:
Ok, I imagined one use case:
```ts
const cookies = useCookies();

watch(() => cookies.get('locale'), () => {
  alert('locale changed', cookies.get('locale'))
})
```
But you can't unsubscribe from `universalCookie` this way, because onMounted/onUnmounted won't work here.

So I would prefer this way:
```ts
const cookies = useCookies()
const locale = cookies.get('locale')

const onChange = () => {
  if (locale !== cookies.get('locale')) {
    locale = cookies.get('locale')
    alert('locale changed', locale)
  }
}

cookies.addChangeListener(onChange)
```

Or return additional method `unmount/deactivate` (for example) from `useCookies()`

Reply:
https://github.com/antfu/vueuse/blob/d13172237d42981a1ccfe760fa7d0c969152c182/packages/core/useEventListener/index.ts#L12-L18

When you use it globally, most of the time you don't need to dispose them.

One other use case is to share the state globally. You can see this comment: https://github.com/antfu/vueuse/pull/150#discussion_r508320224